thunder/core/transforms.py

```diff
@@ -2181,13 +2181,14 @@ def softmax_aug_fwd(a: Proxy, dim: int, dtype: dtypes.dtype | None = None) -> VJ
     from thunder.torch import softmax
 
     primal = softmax(a, dim, dtype=dtype)
-    residuals = (primal, dim)
```

Comment:
I wonder if this would have been easier to spot if we wrote the grad transform for softmax as a single function:

```
fwd = softmax(a, dim, dtype=dtype)

g = get_grad(fwd)
a_grad = fwd * (g - (fwd * g).sum(dim, keepdim=True)
put_grad(a, a_grad)

return fwd
```

I'm not sure. I wonder if `put_grad` should automatically cast the gradient to the dtype of the tensor?